examples/server/main.go

```diff
@@ -0,0 +1,173 @@
+		graphql.ObjectConfig{
+			Name: "Document",
+			Fields: graphql.Fields{
+				"title": &graphql.Field{
+					Type: graphql.String,
+				},
+				"content": &graphql.Field{
+					Type: graphql.String,
+				},
+			},
+		},
+	)
+
+	var queryType = graphql.NewObject(
+		graphql.ObjectConfig{
+			Name: "Query",
+			Fields: graphql.Fields{
+				"document": &graphql.Field{
+					Type: documentType,
+					Args: graphql.FieldConfigArgument{
+						"docId": &graphql.ArgumentConfig{
+							Type: graphql.Int,
+						},
+					},
+					Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+						docID := p.Args["docId"].(int)
+						return documents[docID], nil
+					},
+				},
+			},
+		},
+	)
+
+	var mutationType = graphql.NewObject(
+		graphql.ObjectConfig{
+			Name: "Mutation",
+			Fields: graphql.Fields{
+				"updateDocument": &graphql.Field{
+					Type: documentType,
+					Args: graphql.FieldConfigArgument{
+						"docId": &graphql.ArgumentConfig{
+							Type: graphql.Int,
+						},
+						"title": &graphql.ArgumentConfig{
+							Type: graphql.String,
+						},
+						"content": &graphql.ArgumentConfig{
+							Type: graphql.String,
+						},
+					},
+					Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+
+						docID := p.Args["docId"].(int)
+						documents[docID].Title = p.Args["title"].(string)
+						documents[docID].Title = p.Args["title"].(string)
+
+						for _, subscriptions := range subscriptionManager.Subscriptions() {
+							for _, subscription := range subscriptions {
+
+								// JSON interface is float64
+								var subdocID int = int(subscription.Variables["docId"].(float64))
+
+								if docID == subdocID {
+
+									params := graphql.Params{
+										Schema:         schema,
+										RequestString:  subscription.Query,
+										VariableValues: subscription.Variables,
+										OperationName:  subscription.OperationName,
+									}
+									result := graphql.Do(params)
+
+									data := graphqlws.DataMessagePayload{
+										Data:   result.Data,
+										Errors: graphqlws.ErrorsFromGraphQLErrors(result.Errors),
+									}
+
+									subscription.SendData(subscription, &data)
+								}
+							}
+						}
+
+						return documents[docID], nil
+					},
+				},
+			},
+		},
+	)
+
+	var subscriptionType = graphql.NewObject(
+		graphql.ObjectConfig{
+			Name: "Subscription",
+			Fields: graphql.Fields{
+				"documentUpdates": &graphql.Field{
+					Type: documentType,
+					Args: graphql.FieldConfigArgument{
+						"docId": &graphql.ArgumentConfig{
+							Type: graphql.Int,
+						},
+					},
+					Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+						docID := p.Args["docId"].(int)
+						return documents[docID], nil
+					},
+				},
+			},
+		},
+	)
+
+	schemaConfig := graphql.SchemaConfig{
+		Query:        queryType,
+		Mutation:     mutationType,
+		Subscription: subscriptionType,
+	}
+
+	var err error
+	schema, err = graphql.NewSchema(schemaConfig)
```

Comment:
`schema` is only needed in `main()`, so I'd remove the `var schema graphql.Schema` further towards the top and just do the following here:
```go
schema, err := graphql.NewSchema(schemaConfig)
```
That way you also don't have to declare `err` as a `var` before using it.

Reply:
`schema` is also used in the mutation resolver

Reply:
Ah, you're right, sorry. Ignore this comment then. :)